### PR TITLE
WinCLIP improvements

### PR DESCRIPTION
**Pull request by djdameln**
## 📝 Description

Two small improvements to the WinCLIP torch model:
1. Set device dynamically when collecting text embeddings before inference. This is needed so that `setup` can be called after the model has been moved to cuda.
2. Encode the images without gradient in the forward pass. This is to preserve gpu memory when the model has not been set in `eval` mode by the user.

## ✨ Changes

Select what type of change your PR is:

- [x] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔨 Refactor (non-breaking change which refactors the code base)
- [ ] 🚀 New feature (non-breaking change which adds functionality)
- [ ] 💥 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📚 Documentation update
- [ ] 🔒 Security update

## ✅ Checklist

Before you submit your pull request, please make sure you have completed the following steps:

- [ ] 📋 I have summarized my changes in the [CHANGELOG](https://github.com/openvinotoolkit/anomalib/blob/main/CHANGELOG.md) and followed the guidelines for my type of change (skip for minor changes, documentation updates, and test enhancements).
- [ ] 📚 I have made the necessary updates to the documentation (if applicable).
- [ ] 🧪 I have written tests that support my changes and prove that my fix is effective or my feature works (if applicable).

For more information about code review checklists, see the [Code Review Checklist](https://github.com/openvinotoolkit/anomalib/blob/main/docs/source/markdown/guides/developer/code_review_checklist.md).

### CODE REVIEW

File: src/anomalib/models/image/winclip/torch_model.py [hunk 1]
```diff
@@ -325,15 +326,17 @@ def _collect_text_embeddings(self, class_name: str) -> None:
...
         normal_prompts, anomalous_prompts = create_prompt_ensemble(class_name)
         # tokenize prompts
         normal_tokens = tokenize(normal_prompts)
         anomalous_tokens = tokenize(anomalous_prompts)
         # encode tokens to obtain prompt embeddings
         with torch.no_grad():
-            normal_embeddings = self.clip.encode_text(normal_tokens)
-            anomalous_embeddings = self.clip.encode_text(anomalous_tokens)
+            normal_embeddings = self.clip.encode_text(normal_tokens.to(device))
+            anomalous_embeddings = self.clip.encode_text(anomalous_tokens.to(device))
```
Comment (samet-akcay) [inline src/anomalib/models/image/winclip/torch_model.py:339]: can we get rid of `with torch.no_grad()` here as well?

**Review by samet-akcay — APPROVED** [review]

**Comment by adrianboguszewski** [comment]
Could we also fix #1641 in this PR?
